### ISSUE TITLE
🐛 Bug: eslint does not report problems on `tasks.json` in VS Code

### ISSUE THREAD

**Issue by danvk** [JoshuaKGoldberg/create-typescript-app#1219]
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

I created a new repo yesterday using `create-typescript-app` using the "everything" setting (woo!). I set up a few tasks in VS Code's `tasks.json` and pushed a PR, at which point I noticed that I was getting some eslint errors because of unsorted keys in my `tasks.json`.

These were reported via GitHub Actions CI and when I run `pnpm lint`, but they were not reported in VS Code like lint errors in `.ts` files are.

### Actual

I would expect eslint issues in `tasks.json` to be reported in VS Code (so that I can quick fix them!).

### Additional Info

I noticed that `"json"` is in the `eslint.probe` list in `.code/settings.json`, but maybe it also needs to be told about JSON files some other place?

Here's the PR and repo: https://github.com/danvk/gravlax/pull/3

**Comment by JoshuaKGoldberg** [comment]
@all-contributors please add @danvk for bug.

> 🤖 Beep boop! This comment was added automatically by [all-contributors-auto-action](https://github.com/marketplace/actions/all-contributors-auto-action).
> Not all contributions can be detected from Git & GitHub alone. Please comment any missing contribution types this bot missed.
> ...and of course, thank you for contributing! 💙

**Comment by JoshuaKGoldberg** [comment]
Hmm, looks like using `eslint.probe` for `jsonc` files isn't yet supported in vscode-eslint: https://github.com/microsoft/vscode-eslint/issues/1786. Marking as blocked pending discussion there.

Worst case scenario, we can go back to `eslint.validate`. But that's described as old and legacy so I'd rather not for a bit.

**Comment by JoshuaKGoldberg** [comment]
Good news, looks like this was resolved by https://github.com/microsoft/vscode-eslint/pull/1787! I tried out a straightforward rule violation locally by adding a trailing comma in `.vscode/tasks.json` and it worked:

```ts
$ npx eslint .vscode/tasks.json

/Users/josh/repos/create-typescript-app/.vscode/tasks.json
  7:17  error  Unexpected trailing comma  jsonc/comma-dangle

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

🚀.
